src/gui.cpp

```diff
@@ -502,10 +504,31 @@ bool window_update_controller::check_redraw(GtkWidget *toplevel)
 gui_environment::gui_environment()
 {
     keyfile = g_key_file_new();
+
+    string filename;
     
-    gchar *fn = g_build_filename(getenv("HOME"), ".calfrc", NULL);
-    string filename = fn;
-    g_free(fn);
+    gchar *fn_under_home = g_build_filename(getenv("HOME"), ".calfrc", NULL);
+    char *xdg_conf = getenv("XDG_CONFIG_HOME");
+    gchar *calf_conf_dir = g_build_filename(xdg_conf, "calf", NULL);
+    gchar *fn_under_conf = g_build_filename(calf_conf_dir, "calfrc", NULL);
+
+    //if $HOME/.calfrc exists or XDG_CONFIG_HOME is not set use $HOME/.calfrc
+    if(g_file_test(fn_under_home, G_FILE_TEST_IS_REGULAR) || xdg_conf == NULL) {
+        filename = fn_under_home;
+    } else {
+        //if $XDG_CONFIG_HOME/calf does not exist, create it
+        mkdir(calf_conf_dir, 0644);
```

Comment:
The idea here is that `mkdir` will return `-1` and set `errno` if the directory does not exist? Wouldn't it be cleaner (errno pollution) to prior test if the directory exists, e.g. using `access`?

Reply:
it would indeed be cleaner. I'll check with g_file_test.